packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -118,36 +120,25 @@ export default function markdown({ settings, logger }: AstroPluginOptions): Plug
 
 				${layout ? `import Layout from ${JSON.stringify(layout)};` : ''}
 				import { getImage } from "astro:assets";
+				${imagePaths.map((entry) => `import Astro__${entry.safeName} from ${JSON.stringify(entry.raw)};`)}
 
-				export const images = {
-					${imagePaths.map(
-						(entry) =>
-							`'${entry.raw}': await getImageSafely((await import("${entry.raw}")).default, "${
-								entry.raw
-							}", "${rootRelativePath(settings.config.root, entry.resolved)}")`
-					)}
-				}
-
-				async function getImageSafely(imageSrc, imagePath, resolvedImagePath) {
-					if (!imageSrc) {
-						throw new AstroError({
-							...AstroErrorData.MarkdownImageNotFound,
-							message: AstroErrorData.MarkdownImageNotFound.message(
-								imagePath,
-								resolvedImagePath
-							),
-							location: { file: "${id}" },
-						});
+				const images = async function() {
+					return {
+						${imagePaths
+							.map((entry) => `"${entry.raw}": await getImage({src: Astro__${entry.safeName}})`)
+							.join('\n')}
 					}
-
-					return await getImage({src: imageSrc})
 				}
 
-				function updateImageReferences(html) {
-					return html.replaceAll(
-						/__ASTRO_IMAGE_="([^"]+)"/gm,
-						(full, imagePath) => spreadAttributes({src: images[imagePath].src, ...images[imagePath].attributes})
-					);
+				async function updateImageReferences(html) {
+					return images().then((images) => {
+						return html.replaceAll(/__ASTRO_IMAGE_="([^"]+)"/gm, (full, imagePath) =>
+							spreadAttributes({
+								src: images[imagePath].src,
+								...images[imagePath].attributes,
+							})
+						);
+					});
 				}
 
 				const html = updateImageReferences(${JSON.stringify(html)});
```

Comment:
Since `updateImageReferences` now returns a promise, wouldn't `html` become a promise and affect `compiledContent` and the `Content` render code?

Seems like we need another TLA here but maybe that is able to side-step the Rollup bug.

Reply:
Ah yes, I didn't notice that it made `html` a promise since it worked in rendering... Unfortunately, putting a TLA anywhere in this file causes the issue.

Reply:
I think the Rollup bug is only caused when dynamically importing a module that causes an import loop, but this PR removed the dynamic import as a top-level import, so theoretically, I think we can keep using TLA?

Unless if you're already tested it, then pardon me 😅 We might have to refactor some modules to break the loop otherwise. Or fix in Rollup directly.

Reply:
By TLA, I mean directly on this line, like `const html = await ...`

Reply:
Doing `const html = await` causes the issue, unfortunately

Reply:
I guess I could try fixing this in Rollup directly